types.lua

```diff
@@ -127,6 +127,59 @@ for _,typename in ipairs({"real", "unsigned char", "char", "short", "int", "long
    }
 end
 
+for _,typename in ipairs({"unsigned char", "char", "short", "int", "long"}) do
+   argtypes[typename] = {
+
+      helpname = function(arg)
+                    return typename
+                 end,
+
+      declare = function(arg)
+                   -- if it is a number we initialize here
+                   local default = tonumber(interpretdefaultvalue(arg)) or 0
+                   return string.format("%s arg%d = %g;", typename, arg.i, default)
+                end,
+
+      check = function(arg, idx)
+                 return string.format("lua_isnumber(L, %d)", idx)
+              end,
+
+      read = function(arg, idx)
+                return string.format("arg%d = (%s)lua_tointeger(L, %d);", arg.i, typename, idx)
```

Comment:
wouldn't this break Lua 5.2 and Lua 5.1 support, as they do not have lua_tointeger. Can we guard it with a version check somehow?